### ISSUE TITLE
SGD에서 Stochastic의 의미는?

### ISSUE THREAD

**Issue by jarammm** [AInsteins/AI-Tech-Interview#149]

**Comment by jarammm** [comment]
- 확률적이라는 의미로, sgd는 확률적 경사하강법이라는 뜻이다.
- 경사하강을 할 데이터를 확률적으로 뽑아 진행한다.
- 이는 정확도는 조금 떨어지더라도 보다 빨리, global minima를 찾을 수 있고, local minima에 빠질 위험이 적어진다.

**Comment by jarammm** [comment]
- 확률적이라는 의미로, sgd는 확률적 경사하강법이라는 뜻이다.
- 경사하강을 할 데이터를 확률적으로 뽑아 진행한다.
- 이는 정확도는 조금 떨어지더라도 보다 빨리, global minima를 찾을 수 있고, local minima에 빠질 위험이 적어진다.

**Comment by miinkang** [comment]
- 무작위로 결정된, 이라는 의미
- 전체 데이터셋이 아닌 미니배치 단위로 Loss function을 계산하고
- 미니배치가 무작위로 선택된다는 것을 의미함

**Comment by kimyuuuuuu** [comment]
SGD는 Loss Function을 계산할 때 전체 train dataset을 사용하는 Batch Gradient Descent와 다르게 일부 조그마한 데이터의 모음(mini-batch)에 대해서만 loss function을 계산한다.

`Stochastic`은 **mini-batch가 전체 train dataset에서 무작위로 선택된다**는 것을 의미한다.
